src/libvncclient/tight.c

```diff
@@ -54,24 +54,54 @@
 
 #ifndef RGB_TO_PIXEL
 
-#define RGB_TO_PIXEL(bpp,r,g,b)						\
-  (((CARD##bpp)(r) & client->format.redMax) << client->format.redShift |		\
-   ((CARD##bpp)(g) & client->format.greenMax) << client->format.greenShift |	\
-   ((CARD##bpp)(b) & client->format.blueMax) << client->format.blueShift)
-
-#define RGB24_TO_PIXEL(bpp,r,g,b)                                       \
-   ((((CARD##bpp)(r) & 0xFF) * client->format.redMax + 127) / 255             \
-    << client->format.redShift |                                              \
-    (((CARD##bpp)(g) & 0xFF) * client->format.greenMax + 127) / 255           \
-    << client->format.greenShift |                                            \
-    (((CARD##bpp)(b) & 0xFF) * client->format.blueMax + 127) / 255            \
-    << client->format.blueShift)
-
-#define RGB24_TO_PIXEL32(r,g,b)						\
-  (((uint32_t)(r) & 0xFF) << client->format.redShift |				\
-   ((uint32_t)(g) & 0xFF) << client->format.greenShift |			\
+#define Swap8(c) (c)
+#define Swap16(s) ((((s) & 0xff) << 8) | (((s) >> 8) & 0xff))
```

Comment:
Can [rfbClientSwap16IfLE and friends](https://github.com/LibVNC/libvncserver/blob/master/include/rfb/rfbclient.h#L64) be used instead here?

Reply:
Unfortunately I don't believe so.  `rfbClientSwap16IfLE()` swaps when the host byte order of the client is little-endian, but we want to swap when the host byte order of the client is different from the byte order of the pixel format it is using.  In other words, `rfbClientSwap16IfLE()` will do the right thing only when the pixel format is big-endian, so I don't see a way.

Reply:
If you could add this as a comment in to the new defines, that'd be super nice!

Reply:
Sure, I've updated with a comment!